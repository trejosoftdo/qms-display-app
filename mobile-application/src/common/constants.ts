export const AUTHORIZATION_PENDING_CODE = 'authorization_pending';

export const VALIDATION_ERROR_STATUS_CODE = 400;

export const DEVICE_NOT_CONNECTED_ERROR_MESSAGE = 'Device is not connected';

export const CONNECTION_EXPIRED_ERROR_MESSAGE = 'Connection expired!';

export const APP_PROTO = 'app://';

export const DEVICE_ID_KEY = 'SECURE_DEVICE_ID';

export const CONNECTION_DETAILS_KEY = 'CONNECTION_DETAILS';

export const WEB_PLATFORM = 'web';

export const BEARER_PORTION = 'Bearer';

export const EMPTY_VALUE = '';

export const DEFAULT_REGULAR_FONT_NAME = 'Montserrat-Regular';

export const DEFAULT_LIGHT_FONT_NAME = 'Montserrat-Light';

export const DEFAULT_MEDIUM_FONT_NAME = 'Montserrat-Medium';

export const NORMAL_FONT_WEIGHT = 'normal';

export const ONE_SECOND_MILISECONDS = 1000;

export const INTERVAL_TIME = 10 * ONE_SECOND_MILISECONDS;

export const SERVICES_PATH = '/services';

export const TICKET_DETAILS_PATH = '/ticket-details';

export const HOME_PATH = '/';

export const CATEGORIES_PATH = '/categories';

export const CONFIGURATION_PATH = '/configuration';

export const NO_OP = () => {};


export const LANGUAGES = {
  SPANISH: 'es',
  ENGLISH: 'en',
};

export const CONFIG_OPTIONS = {
  INFORMATION: 'information',
  CONNECTION: 'connection',
  LANGUAGE: 'language',
};
