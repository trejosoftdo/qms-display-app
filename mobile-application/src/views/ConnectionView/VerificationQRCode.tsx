import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import QRCode from 'react-native-qrcode-svg';

/**
 * VerificationQRCodeProps defines the props for the Verification QR Code component.
 *
 * @property verificationURI - Verification URI
 */
interface VerificationQRCodeProps {
  verificationURI?: string;
}

/**
 * A component to display the verification QR code 
 *
 * @param {VerificationQRCodeProps} props - The props for the Verification QR Code component.
 */
const VerificationQRCode: React.FC<VerificationQRCodeProps> = (props: VerificationQRCodeProps) => (
  <View style={styles.container}>
    <QRCode value={props?.verificationURI} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    marginBottom: 32,
  },
});


export default VerificationQRCode;