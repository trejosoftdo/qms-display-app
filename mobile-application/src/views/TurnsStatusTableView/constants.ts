export const BEING_ATTENDED_STATUS_CODE = "BEING_ATTENDED";

export const TO_BE_ATTENDED_STATUS_CODE = "TO_BE_ATTENDED";