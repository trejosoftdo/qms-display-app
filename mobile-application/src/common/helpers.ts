import { Platform } from 'react-native';
import { router } from 'expo-router';
import { APP_PROTO, EMPTY_VALUE, ONE_SECOND_MILISECONDS, WEB_PLATFORM } from './constants';

/** 
 * Checks if the platform is web
 * @returns boolean
 */
export const isWeb = (): boolean => Platform.OS === WEB_PLATFORM;

/** 
 * Gets the current time
 * @returns number
 */
export const getCurrentTime = (): number => new Date().getTime();

/**
 * Calculates the expire time
 * 
 * @param  {number} expiresIn
 * @returns number
 */
export const calculateExpireTime = (expiresIn: number): number => getCurrentTime() + expiresIn * ONE_SECOND_MILISECONDS;


/** 
 * Extracs icon name
 * @returns string
 */
export const extractIconName = (iconUrl: string): string => iconUrl.replace(APP_PROTO, EMPTY_VALUE)

/**
 * Navigates to a given path
 * 
 * @param  {string} pathname
 * @param  {any} params
 * @returns void
 */
export const goToPath = (pathname: string, params: Record<string, any> = undefined): void => {
  router.push({
    pathname,
    params,
  });
};

/**
 * Waits for certain time
 * @param  {number} timeout 
 * @returns Promise<any>
 */
export const delay = (timeout: number) => new Promise((resolve) => {
  setTimeout(resolve, timeout);
});

/**
 * Gets the response blob url
 * @param  {Response} response 
 * @returns Promise<string>
 */
export const getResponseBlobUrl = async (response: Response): Promise<string> => {
  const stream = await response.blob();
  return URL.createObjectURL(stream);
};
