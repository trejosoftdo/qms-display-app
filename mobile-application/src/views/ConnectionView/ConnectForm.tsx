import * as React from 'react';
import { StyleSheet } from 'react-native';
import { Card, TextInput } from 'react-native-paper';
import { useTranslation } from 'react-i18next';
import ActionButton from '../../common/components/ActionButton';
import AppLightTheme from '../../common/theme';
import { EMPTY_VALUE } from '../../common/constants';
import { TRANSLATION_APPLICATION_KEY, TRANSLATION_CONNECT_KEY } from '../../common/translations/translation-keys';


/**
 * ConnectFormProps defines the props for the Connect Form Component.
 *
 * @property onSubmit - form on submit handler
 */
interface ConnectFormProps {
  onSubmit: (applicationId: string) => void;
}

/**
 * A component a connect form
 *
 * @param {ConnectFormProps} props - The props for the Connect Form component.
 */
const ConnectForm: React.FC<ConnectFormProps> = (props: ConnectFormProps) => {
  const { t } = useTranslation();
  const [applicationId, setApplicationId] = React.useState(EMPTY_VALUE);
  return (
    <Card style={styles.container} mode="outlined">
      <Card.Content>
        <TextInput
          style={styles.input}
          label={t(TRANSLATION_APPLICATION_KEY)}
          value={applicationId}
          onChangeText={text => setApplicationId(text)}
          mode="flat"
          left={<TextInput.Icon icon="application-brackets-outline" />}
        />
      </Card.Content>
      <Card.Actions style={styles.action}>
        <ActionButton
          icon="connection"
          message={t(TRANSLATION_CONNECT_KEY)}
          onPress={() => {
            props.onSubmit(applicationId);
          }}
        />
      </Card.Actions>
    </Card>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    maxWidth: 500,
    backgroundColor: AppLightTheme.colors.background,
  },
  action: {
    marginBottom: 10,
    marginTop: 10,
    paddingRight: 16,
  },
  input: {
    backgroundColor: AppLightTheme.colors.background,
  }
});


export default ConnectForm;