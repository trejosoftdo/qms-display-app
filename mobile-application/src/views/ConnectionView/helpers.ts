import { ConnectionData, ConnectionDetails } from "../../common/models";
import { TRANSLATION_CONNECTED_MESSAGE_KEY, TRANSLATION_INITIAL_CONNECT_MESSAGE_KEY, TRANSLATION_SUCCESS_CONNECTION_MESSAGE_KEY, TRANSLATION_TO_CONNECT_MESSAGE_KEY } from "../../common/translations/translation-keys";

/**
 * Gets the correct message key from the data state
 * 
 * @param  {ConnectionData} data
 * @returns string
 */
export const getMessageKey = (data: ConnectionData, connectionDetails?: ConnectionDetails): string => {
  if (connectionDetails?.accessToken) {
    return TRANSLATION_CONNECTED_MESSAGE_KEY;
  }

  if (data?.tokens?.accessToken) {
    return TRANSLATION_SUCCESS_CONNECTION_MESSAGE_KEY;
  }

  return data?.userCode ? TRANSLATION_TO_CONNECT_MESSAGE_KEY : TRANSLATION_INITIAL_CONNECT_MESSAGE_KEY;
};
