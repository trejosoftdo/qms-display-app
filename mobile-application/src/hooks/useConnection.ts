import { useEffect, useState, useRef } from "react";
import { useInterval } from "../common/hooks";
import { ConnectionData, ConnectionDetails, ConnectionResult, DeviceAuthData, DeviceConnectionData, Progress } from "../common/models";
import { connectDevice, getTokensForDevice } from "../common/services/auth";
import { getConnectionDetails, saveConnectionDetails } from "../common/device-connection";
import { calculateExpireTime, getCurrentTime } from "../common/helpers";
import { EMPTY_VALUE, INTERVAL_TIME, ONE_SECOND_MILISECONDS } from "../common/constants";


/**
 * Hook used to connect the device
 * 
 * @returns ConnectionResult
 */
const useConnection = (): ConnectionResult => {
  const [applicationId, setApplicationId] = useState('');
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  const [data, setData] = useState<ConnectionData>({});
  const [connectionDetails, setConnectionDetails] = useState<ConnectionDetails>(undefined);
  const [expireTime, setExpireTime] = useState(null);

  useEffect(() => {
    setLoading(true);
    getConnectionDetails().then(details => {
      if (details) {
        setConnectionDetails(details);
      }
    }).catch((error) => {
      setError(error);
    }).finally(() => {
      setLoading(false);
    })
  }, []);

  useInterval((interval) => {
    if (data?.deviceCode && applicationId && !data?.tokens?.accessToken) {
      getTokensForDevice(applicationId, data?.deviceCode)
        .then((tokens) => {
          setData({
            ...data,
            tokens,
          });

          if (
            tokens.accessToken &&
            tokens.expiresIn &&
            tokens.refreshToken &&
            tokens.refreshExpiresIn
          ) {
            return saveConnectionDetails({
              applicationId,
              deviceCode: data?.deviceCode,
              accessToken: {
                value: tokens.accessToken,
                expiresAt: calculateExpireTime(tokens.expiresIn),
              },
              refreshToken: {
                value: tokens.refreshToken,
                expiresAt: calculateExpireTime(tokens.refreshExpiresIn),
              },
            });
          }
        }).catch((error) => {
          setError(error);
        });
    }

    const currentTime = getCurrentTime();
    if (interval.clear && ((expireTime && currentTime >= expireTime) || data?.tokens?.accessToken)) {
      interval.clear();
    }
  }, INTERVAL_TIME);

  const connect = (appId: string): void => {
    setLoading(true);
    setApplicationId(appId);
    setError(null);
    connectDevice(appId).then(data => {
      setData(data);
      setExpireTime(calculateExpireTime(data.expiresIn));
    }).catch((error) => {
      setError(error);
    }).finally(() => {
      setLoading(false);
    });
  };

  const clear = () => {
    setLoading(false);
    setApplicationId(EMPTY_VALUE);
    setError(null);
    setData({});
    setExpireTime(null);
    setConnectionDetails(undefined);
  };

  return {
    loading,
    error,
    data,
    connect,
    clear,
    connectionDetails,
  };
};

export default useConnection;
