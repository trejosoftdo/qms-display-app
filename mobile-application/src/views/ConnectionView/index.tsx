
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import useConnection from '../../hooks/useConnection';
import { goToPath } from '../../common/helpers';
import {
  ActionButton,
  AppView,
  ConditionalContainer,
  Value,
} from '../../common/components';
import { HOME_PATH } from '../../common/constants';
import { getMessageKey } from './helpers';
import ConnectForm from './ConnectForm';
import VerificationQRCode from './VerificationQRCode';
import { TRANSLATION_RECONNECT_MESSAGE_KEY, TRANSLATION_RETURN_MESSAGE_KEY } from '../../common/translations/translation-keys';


/**
 * ConnectionViewProps defines the props for the Connect Virw Component.
 */
interface ConnectionViewProps {}

/**
 * A component for the connect view
 *
 * @param {ConnectionViewProps} props - The props for the Connect View component.
 */
const ConnectionView: React.FC<ConnectionViewProps> = (props: ConnectionViewProps) => {
  const { t } = useTranslation();
  const {
    data,
    error,
    loading,
    connect,
    clear,
    connectionDetails,
  } = useConnection();
  const messageKey = getMessageKey(data, connectionDetails);
  return (
    <AppView
      headerMessage={t(messageKey)}
      loading={loading}
      error={error}
    >
      <ConditionalContainer display={!!connectionDetails?.deviceCode}>
        <ActionButton
          icon="connection"
          message={t(TRANSLATION_RECONNECT_MESSAGE_KEY)}
          onPress={() => {
            clear();
          }}
        />
      </ConditionalContainer>
      <ConditionalContainer display={!data?.userCode && !connectionDetails?.deviceCode}>
        <ConnectForm
          onSubmit={(applicationId) => {
            connect(applicationId);
          }}
        />
      </ConditionalContainer>
      <ConditionalContainer display={data?.userCode && data?.verificationURI && !data?.tokens?.accessToken}>
        <VerificationQRCode verificationURI={data?.verificationURI} />
        <Value
          value={data?.userCode}
          icon="lock"
        />
      </ConditionalContainer>
      <ConditionalContainer display={!!data?.tokens?.accessToken}>
        <ActionButton
          icon="arrow-left-top"
          message={t(TRANSLATION_RETURN_MESSAGE_KEY)}
          onPress={() => {
            goToPath(HOME_PATH);
          }}
        />
      </ConditionalContainer>
    </AppView>
  );
};

export default ConnectionView;