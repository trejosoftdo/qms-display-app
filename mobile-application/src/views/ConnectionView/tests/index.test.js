import React from 'react';
import { act } from '@testing-library/react-hooks';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { goToPath } from '../../../common/helpers';
import { HOME_PATH } from '../../../common/constants';
import useConnection from '../../../hooks/useConnection';
import { getMessageKey } from '../helpers';
import ConnectionView from '../';


jest.mock('../../../common/helpers');
jest.mock('../../../hooks/useConnection');
jest.mock('../helpers');


describe('ConnectionView component', () => {
  const mockMessageKey = 'mock-message-key';

  beforeEach(() => {
    getMessageKey.mockReturnValue(mockMessageKey);
  });

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('renders the component as expected when loading', () => {
    useConnection.mockReturnValue({ loading: true });
    expect(renderShallow(<ConnectionView />)).toMatchSnapshot();
    expect(getMessageKey).toHaveBeenCalledTimes(1);
    expect(getMessageKey).toHaveBeenCalledWith(undefined, undefined);
    expect(useConnection).toHaveBeenCalledTimes(1);
    expect(useConnection).toHaveBeenCalledWith();
  });

  it('renders the component as expected when having the device code', () => {
    const connectionDetails = { deviceCode: 'mock-device-code' };
    useConnection.mockReturnValue({ loading: true, connectionDetails });
    expect(renderShallow(<ConnectionView />)).toMatchSnapshot();
    expect(getMessageKey).toHaveBeenCalledTimes(1);
    expect(getMessageKey).toHaveBeenCalledWith(undefined, connectionDetails);
    expect(useConnection).toHaveBeenCalledTimes(1);
    expect(useConnection).toHaveBeenCalledWith();
  });

  it('renders the component as expected when having the device code, user code and verification uri', () => {
    const connectionDetails = { deviceCode: 'mock-device-code' };
    const data = {
      userCode: 'mock-user-code',
      verificationURI: 'https://test-verification-uri.test',
    };
    useConnection.mockReturnValue({ loading: true, data, connectionDetails });
    expect(renderShallow(<ConnectionView />)).toMatchSnapshot();
    expect(getMessageKey).toHaveBeenCalledTimes(1);
    expect(getMessageKey).toHaveBeenCalledWith(data, connectionDetails);
    expect(useConnection).toHaveBeenCalledTimes(1);
    expect(useConnection).toHaveBeenCalledWith();
  });

  it('renders the component as expected when having the device code, user code, verification uri and access token', () => {
    const connectionDetails = { deviceCode: 'mock-device-code' };
    const data = {
      userCode: 'mock-user-code',
      verificationURI: 'https://test-verification-uri.test',
      tokens: {
        accessToken: 'mock-access-token',
      },
    };
    useConnection.mockReturnValue({ loading: true, data, connectionDetails });
    expect(renderShallow(<ConnectionView />)).toMatchSnapshot();
    expect(getMessageKey).toHaveBeenCalledTimes(1);
    expect(getMessageKey).toHaveBeenCalledWith(data, connectionDetails);
    expect(useConnection).toHaveBeenCalledTimes(1);
    expect(useConnection).toHaveBeenCalledWith();
  });

  it('clear everything when clicking reconnect', () => {
    const mockClear = jest.fn();
    const connectionDetails = { deviceCode: 'mock-device-code' };
    
    useConnection.mockReturnValue({ loading: true, connectionDetails, clear: mockClear });
    const { getByText } = render(<ConnectionView />);

    const element = getByText('Translated[translation:reconnect]({})');
    
    fireEvent.press(element);

    expect(mockClear).toHaveBeenCalledTimes(1);
    expect(mockClear).toHaveBeenCalledWith();
    expect(getMessageKey).toHaveBeenCalledTimes(1);
    expect(getMessageKey).toHaveBeenCalledWith(undefined, connectionDetails);
    expect(useConnection).toHaveBeenCalledTimes(1);
    expect(useConnection).toHaveBeenCalledWith();
  });

  it('connects the application as expected', async () => {
    const mockApplicationId = 'mock-application-id';
    const mockConnect = jest.fn();
    const connectionDetails = { deviceCode: 'mock-device-code' };
    
    useConnection.mockReturnValue({ loading: true, connect: mockConnect });
    const { getByTestId } = render(<ConnectionView />);
    
    const input = getByTestId('text-input-flat');
    const submitButton = getByTestId('button');
    
    act(() => {
      fireEvent.changeText(input, mockApplicationId);
    });

    await waitFor(() => {
      expect(input.props.value).toEqual(mockApplicationId);
    });
  
    act(() => {
      fireEvent.press(submitButton);
    });

    expect(mockConnect).toHaveBeenCalledTimes(1);
    expect(mockConnect).toHaveBeenCalledWith(mockApplicationId);
    expect(getMessageKey).toHaveBeenCalledTimes(2);
    expect(getMessageKey).toHaveBeenCalledWith(undefined, undefined);
    expect(useConnection).toHaveBeenCalledTimes(2);
    expect(useConnection).toHaveBeenCalledWith();
  });

  it('returns to home when clicking return button', async () => {
    const connectionDetails = { deviceCode: 'mock-device-code' };
    const data = {
      userCode: 'mock-user-code',
      verificationURI: 'https://test-verification-uri.test',
      tokens: {
        accessToken: 'mock-access-token',
      },
    };
    useConnection.mockReturnValue({ loading: true, data, connectionDetails });

    const { getByText } = render(<ConnectionView />);
    
    const submitButton = getByText('Translated[translation:return]({})');
    
    act(() => {
      fireEvent.press(submitButton);
    });

    expect(goToPath).toHaveBeenCalledTimes(1);
    expect(goToPath).toHaveBeenCalledWith(HOME_PATH);
    expect(getMessageKey).toHaveBeenCalledTimes(1);
    expect(getMessageKey).toHaveBeenCalledWith(data, connectionDetails);
    expect(useConnection).toHaveBeenCalledTimes(1);
    expect(useConnection).toHaveBeenCalledWith();
  });
});
