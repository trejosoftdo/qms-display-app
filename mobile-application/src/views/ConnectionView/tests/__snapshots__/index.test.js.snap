// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionView component renders the component as expected when having the device code 1`] = `
<AppView
  headerMessage="Translated[mock-message-key]({})"
  loading={true}
>
  <ConditionalContainer
    display={true}
  >
    <ActionButton
      icon="connection"
      message="Translated[translation:reconnect]({})"
      onPress={[Function]}
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={false}
  >
    <ConnectForm
      onSubmit={[Function]}
    />
  </ConditionalContainer>
  <ConditionalContainer>
    <VerificationQRCode />
    <Value
      icon="lock"
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={false}
  >
    <ActionButton
      icon="arrow-left-top"
      message="Translated[translation:return]({})"
      onPress={[Function]}
    />
  </ConditionalContainer>
</AppView>
`;

exports[`ConnectionView component renders the component as expected when having the device code, user code and verification uri 1`] = `
<AppView
  headerMessage="Translated[mock-message-key]({})"
  loading={true}
>
  <ConditionalContainer
    display={true}
  >
    <ActionButton
      icon="connection"
      message="Translated[translation:reconnect]({})"
      onPress={[Function]}
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={false}
  >
    <ConnectForm
      onSubmit={[Function]}
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={true}
  >
    <VerificationQRCode
      verificationURI="https://test-verification-uri.test"
    />
    <Value
      icon="lock"
      value="mock-user-code"
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={false}
  >
    <ActionButton
      icon="arrow-left-top"
      message="Translated[translation:return]({})"
      onPress={[Function]}
    />
  </ConditionalContainer>
</AppView>
`;

exports[`ConnectionView component renders the component as expected when having the device code, user code, verification uri and access token 1`] = `
<AppView
  headerMessage="Translated[mock-message-key]({})"
  loading={true}
>
  <ConditionalContainer
    display={true}
  >
    <ActionButton
      icon="connection"
      message="Translated[translation:reconnect]({})"
      onPress={[Function]}
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={false}
  >
    <ConnectForm
      onSubmit={[Function]}
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={false}
  >
    <VerificationQRCode
      verificationURI="https://test-verification-uri.test"
    />
    <Value
      icon="lock"
      value="mock-user-code"
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={true}
  >
    <ActionButton
      icon="arrow-left-top"
      message="Translated[translation:return]({})"
      onPress={[Function]}
    />
  </ConditionalContainer>
</AppView>
`;

exports[`ConnectionView component renders the component as expected when loading 1`] = `
<AppView
  headerMessage="Translated[mock-message-key]({})"
  loading={true}
>
  <ConditionalContainer
    display={false}
  >
    <ActionButton
      icon="connection"
      message="Translated[translation:reconnect]({})"
      onPress={[Function]}
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={true}
  >
    <ConnectForm
      onSubmit={[Function]}
    />
  </ConditionalContainer>
  <ConditionalContainer>
    <VerificationQRCode />
    <Value
      icon="lock"
    />
  </ConditionalContainer>
  <ConditionalContainer
    display={false}
  >
    <ActionButton
      icon="arrow-left-top"
      message="Translated[translation:return]({})"
      onPress={[Function]}
    />
  </ConditionalContainer>
</AppView>
`;
